cw-orch-daemon/src/env.rs

```diff
@@ -0,0 +1,124 @@
+    /// `folder/file.json` will resolve to `~/.cw-orchestrator/folder/file.json`
+    /// `./folder/file.json` will resolve `$pwd/folder/file.json`
+    /// `../folder/file.json` will resolve `$pwd/../folder/file.json`
+    /// `/usr/var/file.json` will resolve to `/usr/var/file.json`
+    /// Defaults to "~./cw-orchestrator/state.json"
+    pub state_file: PathBuf,
+
+    /// Optional - Float
+    /// This allows changing the gas buffer applied after tx simulation
+    /// If not specified, a more complex algorithm is applied for dealing with small gas fee cases
+    pub gas_buffer: Option<f64>,
+
+    /// Optional - Integer
+    /// Defaults to None
+    /// Minimum gas amount. Useful when transaction still won't pass even when setting a high gas_buffer or for mixed transaction scripts
+    pub min_gas: Option<u64>,
+
+    /// Optional - Integer
+    /// Defaults to [`DEFAULT_TX_QUERY_RETRIES`]
+    /// This changes the number of tx queries before it fails if it doesn't find any result
+    pub max_tx_query_retries: usize,
+
+    /// Optional - Integer
+    /// Defaults to 1
+    /// Minimum block speed in seconds. Useful when the block speeds are varying a lot
+    pub min_block_speed: u64,
+
+    /// Optional - boolean
+    /// Defaults to "false"
+    /// Disable wallet balance assertion.
+    /// When balance assertion is enabled, it asserts that the balance of the sender is sufficient before submitting any transactions (during the simulation step)
+    pub disable_wallet_balance_assertion: bool,
+
+    /// Optional - boolean
+    /// Defaults to "false"
+    /// Disable the "Enable Logs" message
+    /// It allows forcing cw-orch to not output anything
+    pub disable_logs_message: bool,
+}
+
+/// Fetches the default state folder.
+/// This function should only error if the home_dir is not set and the `dirs` library is unable to fetch it
+/// This happens only in rare cases
+pub fn default_state_folder() -> Result<PathBuf, StdError> {
+    dirs::home_dir().map(|home| home.join(".cw-orchestrator"))
+        .ok_or( StdError::generic_err(
+            format!(
+                "Your machine doesn't have a home folder. You can't use relative path for the state file such as 'state.json'. 
+                Please use an absolute path ('/home/root/state.json') or a dot-prefixed-relative path ('./state.json') in the {} env variable.",
+                STATE_FILE_ENV_NAME
+            )))
+}
+
+impl Default for DaemonEnvVars {
+    fn default() -> Self {
+        DaemonEnvVars {
+            state_file: PathBuf::from_str("state.json").unwrap(),
+            gas_buffer: None,
+            min_gas: None,
+            max_tx_query_retries: DEFAULT_TX_QUERY_RETRIES,
+            min_block_speed: 1,
+            disable_wallet_balance_assertion: false,
+            disable_logs_message: false,
+        }
+    }
+}
+
+impl DaemonEnvVars {
+    pub fn load() -> Result<Self, CwEnvError> {
+        let mut env_values = DaemonEnvVars::default();
+
```

Comment:
Agreed, would be nice to use cell here.

Reply:
Using OnceCell or lazy_static here means it's not possible to change those env variables in code, once it's initialized. 
Because this is hidden from the user, unexpected behavior can happen which would be hard to debug for the user. 
I would prefer the second solution !

Reply:
I wouldn't lazy static it but initialize it with the daemon state instead. That way the user can still execute code (and set envs) before this is loaded.